pkg/rpc/dial.go

```diff
@@ -14,8 +15,10 @@ import (
 	"github.com/prysmaticlabs/prysm/v4/network/authorization"
 )
 
+var dialError = "Dial ethclient error: "
```

Comment:
Emm since its just a string type value but not an error type, name `dialError` makes me feeling confusing, i think its ok to return the raw error in `DialClientWithBackoff`, since we have already had a `log.Error("Dial ethclient error", "url", url, "error", err)`.